jenkins/databricks/test.sh

```diff
@@ -36,11 +61,37 @@ sudo "$(which pip)" install pytest sre_yield requests pandas pyarrow findspark p
 export SPARK_HOME=/databricks/spark
 # change to not point at databricks confs so we don't conflict with their settings
 export SPARK_CONF_DIR=$PWD
-export PYTHONPATH=$SPARK_HOME/python:$SPARK_HOME/python/pyspark/:$SPARK_HOME/python/lib/py4j-0.10.9-src.zip
-if [[ $BASE_SPARK_VERSION == "3.2.1" ]]
+# Get Python version from environment after lowercase and remove minor.
+# Something like python3.8 for DB10.4 and python3.9 for DB11.3
+sw_versions[PYTHON]=$(python --version | sed 's/[A-Z]/\L&/g;s/ //g' | cut -d . -f 1,2)
+# set Iceberg related versions. See https://iceberg.apache.org/multi-engine-support/#apache-spark
+# Spark 3.3 | Initial Iceberg Version 0.14.0 | Latest Support 1.0.0
+# Spark 3.2 | Initial Iceberg Version 0.13.0 | Latest Support 1.0.0
+# Spark 3.1 | Initial Iceberg Version 0.12.0 | Latest Support 1.0.0
+case "$BASE_SPARK_VERSION" in
+    "3.3.0")
+        sw_versions[ICEBERG]=${ICEBERG_VERSION:-'0.14.1'}
```

Comment:
we don't support 0.14.1... its fine to set it here I guess but at the same time wouldn't expect it to work so could confuse people. so seems like it should be skipped until we do

Reply:
Thanks @tgravescs ,
Then the default should be set to '0.14.0' to make it easy for manual testing.
According to https://github.com/NVIDIA/spark-rapids/pull/6098 Jenkins sets ICEBERG_VERSION to `0.14.0`

Reply:
#6104 reverted #6098.  There should be nothing referencing 0.14.0.

Reply:
> I updated run_iceberg_tests to skip iceberg for spark-3.4+ instead of spark-3.3+

Have you tested that change?  We compile against Iceberg 0.13, and that is not binary compatible with Iceberg 0.14, the latter which is required to run on Spark 3.3.x.

Reply:
Yeah, it is fine that the build does not support 0.14.
@gerashegalov and I decided to make the changes related to the profiler build of DB11.3 in [PR-7152: Add a shim for Databricks 11.3. Spark 3.3.0](https://github.com/NVIDIA/spark-rapids/pull/7152).
We need to be able to run the tests to continue fixing the new shims.